pkg/infra/webhooks.go

```diff
@@ -112,6 +113,21 @@ var _ = Describe("Webhooks", framework.LabelMAPI, framework.LabelDisruptive, fun
 			if err != nil {
 				return err
 			}
+
+			failed := framework.FilterMachines([]*machinev1beta1.Machine{m}, framework.MachinePhaseFailed)
+			if len(failed) > 0 {
+				reason := "failureReason not present in Machine.status"
+				if m.Status.ErrorReason != nil {
+					reason = string(*m.Status.ErrorReason)
+				}
+				message := "failureMessage not present in Machine.status"
+				if m.Status.ErrorMessage != nil {
+					message = *m.Status.ErrorMessage
+				}
+				klog.Errorf("Failed machine: %s, Reason: %s, Message: %s", m.Name, reason, message)
+			}
+			Expect(len(failed)).To(Equal(0), "zero machines should be in a Failed phase")
```

Comment:
You could also achieve this with, I don't have a strong opinion either way
```
Expect(failed).ToNot(ContainElement(HaveField("Status", SatisfyAny(
    HaveField("ErrorReason", Not(BeNil)),
    HaveField("ErrorMessage", Not(BeNil)),
))))
```

Reply:
Got it, thanks. This time I will keep things as they are.